sdk/servicebus/service-bus/src/client.ts

```diff
@@ -25,3 +25,8 @@ export interface Client {
    */
   detached(error?: AmqpError | Error): Promise<void>;
 }
+
+/**
+ * @internal
+ */
+export type ClientType = "QueueClient" | "TopicClient" | "SubscriptionClient";
```

Comment:
Why not use enums instead? Will help avoid typo errors in future https://www.typescriptlang.org/docs/handbook/enums.html

Reply:
I don't think it helps with typos. TS users will get the same checking either way, and the tooling is roughly equivalent (users can still tab-complete string literals). JS users will just pass string literals as they're used to. This method has the benefit of documenting things closer to how JS users will understand things.

I am an enum skeptic though, so take this advice in that light.

Reply:
@bterlson This is only used internally in the SDK, it is not exposed to the users

I am fine with using enum. Updated with https://github.com/Azure/azure-sdk-for-js/pull/2207/commits/78a9bc00996775d394569974a977e7df430374b7

Reply:
I was speaking generally about how I see various developers working with this codebase. In general, enums don't help and are often annoying IMO.